### PR TITLE
[x86/Linux] Fix unsupported architecture in seh-unwind.cpp

### PR DESCRIPTION
**Pull request by seanshpark**
Fix compile error for x86/Linux
- add ASSIGN_UNWIND_REGS in seh-unwind.cpp for x86
- add CONTEXT_EXCEPTION_ACTIVE in pal.h for x86
- add CONTEXT_XSTATE in pal.h for x86

### CODE REVIEW

**Comment by seanshpark** [comment]
I don't know much about unwind so please guide me if anything needs to be fixed in this patch.

File: src/pal/src/exception/seh-unwind.cpp [hunk 1]
```diff
@@ -122,6 +128,11 @@ static void WinContextToUnwindCursor(CONTEXT *winContext, unw_cursor_t *cursor)
     unw_set_reg(cursor, UNW_X86_64_R13, winContext->R13);
     unw_set_reg(cursor, UNW_X86_64_R14, winContext->R14);
     unw_set_reg(cursor, UNW_X86_64_R15, winContext->R15);
+#elif defined(_X86_)
+    unw_set_reg(cursor, UNW_REG_IP, winContext->Eip);
```
Comment (janvorli) [inline src/pal/src/exception/seh-unwind.cpp]: I am sorry I have not noticed this before, but we need the Rsi and Rdi here as well (and the same in the UnwindContextToWinContext below)

Comment (seanshpark) [inline src/pal/src/exception/seh-unwind.cpp]: Thank you :) It's OK to change again. This is a huge place.

**Comment by seanshpark** [comment]
Done adding `Esi` and `Edi`.
And also added `Esi`, `Edi` to `ASSIGN_UNWIND_REGS` macro, but if not needed I'll remove them.

**Comment by seanshpark** [comment]
@dotnet-bot test Windows_NT x86 compatjit Checked Build and Test please

**Pull request merged**